nautobot/extras/models/metadata.py

```diff
@@ -135,3 +141,249 @@ def save(self, *args, **kwargs):
+                    #     ):
+                    #         raise ValidationError(
+                    #             f"Value must not exceed {self.metadata_type.validation_maximum} characters in length"
+                    #         )
+                    #     if self.metadata_type.validation_regex and not re.search(
+                    #         self.metadata_type.validation_regex, json_value
+                    #     ):
+                    #         raise ValidationError(f"Value must match regex '{self.metadata_type.validation_regex}'")
+
+                # Validate integer
+                elif metadata_type_data_type == MetadataTypeDataTypeChoices.TYPE_INTEGER:
+                    try:
+                        value = int(value)
+                    except ValueError:
+                        raise ValidationError("Value must be an integer.")
+                    # TODO uncomment this when MetaDataType validation_minimum, validation_maximum and validation_regex fields are implemented
+                    # if self.metadata_type.validation_minimum is not None and value < self.metadata_type.validation_minimum:
+                    #     raise ValidationError(f"Value must be at least {self.metadata_type.validation_minimum}")
+                    # if self.metadata_type.validation_maximum is not None and value > self.metadata_type.validation_maximum:
+                    #     raise ValidationError(f"Value must not exceed {self.metadata_type.validation_maximum}")
+                # Validate integer
+                elif metadata_type_data_type == MetadataTypeDataTypeChoices.TYPE_FLOAT:
+                    try:
+                        value = float(value)
+                    except ValueError:
+                        raise ValidationError("Value must be an float.")
+                # Validate boolean
+                elif metadata_type_data_type == MetadataTypeDataTypeChoices.TYPE_BOOLEAN:
+                    try:
+                        value = is_truthy(value)
+                    except ValueError as exc:
+                        raise ValidationError("Value must be true or false.") from exc
+
+                # Validate date
+                elif metadata_type_data_type == MetadataTypeDataTypeChoices.TYPE_DATE:
+                    if not isinstance(value, date):
+                        try:
+                            datetime.strptime(value, "%Y-%m-%d")
+                        except TypeError:
+                            raise ValidationError("Value must be a date or str object.")
+                        except ValueError:
+                            raise ValidationError("Date values must be in the format YYYY-MM-DD.")
+                # Validate datetime
+                elif metadata_type_data_type == MetadataTypeDataTypeChoices.TYPE_DATETIME:
+                    if not isinstance(value, datetime):
+                        try:
+                            datetime.strptime(value, "%Y-%m-%d %H:%M:%S")
```

Comment:
Let's be sure to test the round-trip behavior of this field type, especially when timezone-aware datetimes are involved? Let's also make sure the string format here matches JSON conventions for storing timezone-aware datetimes.